src/wp-includes/class-wp-application-passwords.php

```diff
@@ -58,15 +58,24 @@ public static function is_in_use() {
 	 * Creates a new application password.
 	 *
 	 * @since 5.6.0
+	 * @since 5.7.0 Returns WP_Error if application name already exists.
 	 *
 	 * @param int   $user_id  User ID.
 	 * @param array $args     Information about the application password.
 	 * @return array|WP_Error The first key in the array is the new password, the second is its detailed information.
 	 *                        A WP_Error instance is returned on error.
 	 */
 	public static function create_new_application_password( $user_id, $args = array() ) {
+		if ( ! empty( $args['name'] ) ) {
+			$args['name'] = sanitize_text_field( $args['name'] );
+		}
+
 		if ( empty( $args['name'] ) ) {
-			return new WP_Error( 'application_password_empty_name', __( 'An application name is required to create an application password.' ) );
+			return new WP_Error( 'application_password_empty_name', __( 'An application name is required to create an application password.' ), array( 'status' => 400 ) );
+		}
+
+		if ( self::user_application_name_exists( $user_id, $args['name'] ) ) {
+			return new WP_Error( 'application_password_unique_name', __( 'An application name should be unique to create an application password.' ), array( 'status' => 409 ) );
```

Comment:
```suggestion
			return new WP_Error( 'application_password_duplicate_name', __( 'Each application name should be unique.' ), array( 'status' => 409 ) );
```

I think typically WP Error codes express the negative form instead of the positive.

Reply:
I wondered about that too. Thanks for the suggestion. Applying it.